ZodiacVision/main.py

```diff
@@ -55,3 +28,50 @@
     #     streamer.write(mask)
     # else:
     #     streamer.write(stream_image)
+
+class VisionDispatcher(socketserver.BaseRequestHandler):
+    def handle(self):
+        mask = self.getMask()
+        center_x, center_y, distance = -1, -1, -1
+        if mask.all() != -1: # if -1 - give them that, let the client handle
+            contour, center_x, center_y = detector.findTarget(mask)
+            distance = camera.findTargetDistance(center_x, center_y)
+
+        msg = self.request.recv(1024).strip()
+        self.dispatchResponse(msg, center_x, center_y, distance)
+        fpsCounter.update()
+        fpsCounter.stop()
+
+    def reply(self, msg):
+        self.request.sendall(msg.encode())
+
+    def dispatchResponse(self, msg, center_x, center_y, distance):
+        if msg == 'distance':
+            self.reply(("distance|"+distance+"\n"))
+        elif msg == 'center_x':
+            self.reply("center_x|"+center_x+"\n")
+        elif msg == 'cxy':
+            self.reply(""+center_x+","+center_y)
+        elif msg == 'cxd':
+            self.reply(""+center_x+","+center_y+","+distance)
+        else:
+            self.reply('error|invalid request\n')
+        
+    def getMask():
+        fpsCounter.reset()
+        fpsCounter.start()
+        if net.update_exposure:
+            visionCamera.updateExposure(net)
+            net.update_exposure = False
+        # A new image is available if grab() returns SUCCESS
+        frame = visionCamera.read()
+        if net.vision_use:
+            cv2.imwrite('/home/jetson/ZodiacVision/capture/' + time.strftime("%Y%m%d-%H%M%S") + '.png', frame)
+            net.vision_use = False
+        return detector.preProcessFrame(frame)
+
+
+port = net.yml_data['main']['port']
+
+with socketserver.TCPServer(('localhost', port), VisionDispatcher) as server:
```

Comment:
Note that this is blocking, according to the docs. May be better to use a threaded one? Not sure yet.